In [1]:
import numpy as np
from scipy.stats import norm

mu = 0     # Media
sigma = 1  # Desviación estándar
N = 1000   # Tamaño de la muestra

# Generación de muestra teórica
muestra_teorica = norm.rvs(mu, sigma, size=N)


In [2]:
# Generación de muestra empírica usando transformada inversa
U = np.random.rand(N)  # Números uniformemente distribuidos
muestra_empirica = mu + sigma * norm.ppf(U)


In [3]:
from scipy.stats import chisquare

# Definir las clases (valores posibles) para el histograma
valores = np.linspace(min(muestra_teorica.min(), muestra_empirica.min()), 
                      max(muestra_teorica.max(), muestra_empirica.max()), 
                      20)

# Calcular las frecuencias observadas y esperadas
freq_teo, _ = np.histogram(muestra_teorica, bins=valores)
freq_emp, _ = np.histogram(muestra_empirica, bins=valores)

# Aplicar la prueba Chi-Cuadrado
chi2, p_chi = chisquare(freq_emp, f_exp=freq_teo)
print(f"Chi2 = {chi2:.4f}, p-valor = {p_chi:.4f}")


Chi2 = 37.7523, p-valor = 0.0042


In [4]:
from scipy.stats import ks_2samp

# Aplicar la prueba Kolmogorov-Smirnov
ks_stat, p_ks = ks_2samp(muestra_teorica, muestra_empirica)
print(f"KS = {ks_stat:.4f}, p-valor = {p_ks:.4f}")


KS = 0.0440, p-valor = 0.2878


Las pruebas realizadas sobre las muestras generadas de una distribución normal muestran resultados contradictorios. La prueba de Chi-Cuadrado, con un p-valor de 0.0042, sugiere que las muestras no provienen de la misma distribución, ya que el p-valor es menor que 0.05. Sin embargo, la prueba de Kolmogorov-Smirnov, con un p-valor de 0.2878, no rechaza la hipótesis nula, indicando que no hay diferencias significativas entre las distribuciones acumuladas. En conclusión, aunque la prueba de Kolmogorov-Smirnov sugiere que las distribuciones acumuladas son similares, la prueba de Chi-Cuadrado muestra diferencias importantes, lo que lleva a concluir que las muestras no pueden considerarse como provenientes de la misma distribución. 